### ISSUE TITLE
support for vite

### ISSUE THREAD

**Issue by baseplate-admin** [r0x0r/pywebview#1301]
### Description
It would be awesome if `pywebview` provided support for first class bundlers (eg: `webpack`,`vite`)
 
### Practicalities

- YES/NO I am willing to work on this issue myself.

**Comment by r0x0r** [comment]
How would bundler support look like?

**Comment by baseplate-admin** [comment]
Hi sorry for late reply.

> How would bundler support look like?

Something like tauri, where we use `vite`  to bundle assets.

Why?
1. This would allow us to the pre-processed js frameworks like `solid.js`, `svelte`, `vue`
2. Generally working with bundlers are easier compared to manual management.

Outcome:
1. One single `js` file that allows users to access the gui.

**Comment by r0x0r** [comment]
Tauri has a node application that creates a project scaffold with a configuration file. Is this what you suggest?

**Comment by baseplate-admin** [comment]
Correct. We use node.js to make a bundle. Final application wont have any node.js dependency

**Comment by r0x0r** [comment]
While it is a noble idea, but the reality is that I do not have resources for launching and maintaining a separate project.

**Comment by baseplate-admin** [comment]
> While it is a noble idea, but the reality is that I do not have resources for launching and maintaining a separate project.

Fair enough, but could you please keep this issue open? Maybe someone will implement that feature

**Comment by r0x0r** [comment]
Sure

**Comment by thecircleisround** [comment]
I've updated the react boilerplate to use vite and vue3 here [Vu3 Boilerplate](https://github.com/thecircleisround/pywebview-vue3-boilerplate). I have not tested on windows. Should still work with react.

**Comment by Barafu** [comment]
One does not need any support from PyWebView to use it with Vite. For develop, just start Vite: `npx vite dev` and it will print a path to pass to PyWebView as URL. For release, run `npx vite build` and bundle that folder with your app. 

I am using PyWebView, Vite and PyInstaller together in my app. If you want, look at [this particular ](https://github.com/Barafu/Dream_Spinner/commit/d2b2b839d7ff31b9c208f6583465a9f5bd4fbe7b) state where everything works together. A lot of magic happens in `.vscode/tasks.json` which is why I committed it to Git when usually it is not done.
